### ISSUE TITLE
New factor levels in step_dummy

### ISSUE THREAD

**Issue by EdwinTh** [tidymodels/recipes#159]
New levels in a factor variable now go silently to NA when using `step_dummy`

```r
library(recipes)
a <- data.frame(x = factor(letters[1:3]))
b <- data.frame(x = factor(letters[1:4]))

recipe(a) %>% 
  step_dummy(x) %>% 
  prep() %>% 
  bake(b)
```

I think a warning should be thrown when it is introducing NA values.

Moreover, I think a `check_new_levels` would be a nice addition. It would break `bake` if new factor levels are introduced. What do you reckon?

**Comment by topepo** [comment]
> I think a warning should be thrown when it is introducing NA values.

the current github version  does:

```r
> a <- data.frame(x = factor(letters[1:3]))
> b <- data.frame(x = factor(letters[1:4]))
> 
> recipe(a) %>% 
+     step_dummy(x) %>% 
+     prep() %>% 
+     bake(b)
# A tibble: 4 x 2
    x_b   x_c
  <dbl> <dbl>
1    0.    0.
2    1.    0.
3    0.    1.
4   NA    NA 
Warning message:
There are new levels in a factor: d 
```

**Comment by EdwinTh** [comment]
Great, apparently I was on the CRAN version, not on the dev version. 

Still I think a check would make a good addition, because 
(a) you can miss the warning and proceed, check is hard 
(b) you can also not be using `step_dummy` and still wanting to do the check. (For instance something that is already numerically coded, or using methods that can handle categoricals such as trees).

**Comment by alexpghayes** [comment]
For one-hot encodings, it's also fairly standard to set all dummies to zero for novel levels.

**Comment by EdwinTh** [comment]
So a check on top on the encoding would be a good thing, right?

**Comment by topepo** [comment]
I think that your original proposal would be fine.

**Comment by EdwinTh** [comment]
Will start working on it as soon as job and baby allow :)...

**Comment by BigTimeStats** [comment]
> 
> 
> For one-hot encodings, it's also fairly standard to set all dummies to zero for novel levels.

Agreed, was my point in: https://github.com/tidymodels/recipes/issues/105

**Comment by topepo** [comment]
`step_novel` doesn't make the dummy variables; it just adds another level so it cannot affect the creation of the indicators (one-hot-or-not).

**Comment by EdwinTh** [comment]
This is closed by #363
